src/util/mpir_nodemap.c

```diff
@@ -398,7 +398,7 @@ UT_icd hostname_icd = { MAX_HOSTNAME_LEN, NULL, NULL, NULL };
 /* Lookup node_id in MPIR_Process.node_hostnames, insert and return next_node_id if it's new hostname */
 int MPIR_nodeid_lookup(const char *hostname, int *node_id)
 {
-    if (MPIR_pmi_has_local_cliques()) {
+    if (!MPIR_Process.node_hostnames) {
```

Comment:
are we losing something by not also checking `MPIR_pmi_has_local_cliques()` after this change? can we not have hostnames but still do local cliques?

Reply:
`node_hostnames` will remain NULL if cliques are enabled.  Effectively, `node_hostnames` serve as flags for whether the node_id_init is done.

Reply:
ok, got it